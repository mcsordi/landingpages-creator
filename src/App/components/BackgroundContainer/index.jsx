import P from 'prop-types';
function BackgroundContainer({ children, darkBg = false, image = false }) {
  return (
    <section
      className={`${darkBg ? `bg-[#0A1128]` : 'bg-white'} flex flex-col w-full min-h-screen`}
    >
      {image && (
        <img
          className="absolute w-screen h-screen object-cover max-h-screen"
          alt="Background Container Image"
          src="./images/main-bg.svg"
        />
      )}
      <div
        className={`${darkBg || image ? `text-white` : `text-primary`} relative mx-auto my-auto max-w-[1200px] px-8 py-6`}
      >
        {children}
      </div>
    </section>
  );
}

BackgroundContainer.propTypes = {
  children: P.node,
  darkBg: P.bool,
  image: P.bool,
};
export default BackgroundContainer;
