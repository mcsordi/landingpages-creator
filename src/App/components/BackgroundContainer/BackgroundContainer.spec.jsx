import { render } from '@testing-library/react';
import BackgroundContainer from '.';

describe('<BackgroundContainer/>', () => {
  it('should render BackgroundContainer Default', () => {
    const container = render(<BackgroundContainer />);
    expect(container.container.firstChild).toHaveClass(
      'bg-white flex flex-col w-full min-h-screen',
    );
    expect(container.container.firstChild.firstChild).toHaveClass(
      'relative mx-auto my-auto max-w-[1200px] p-8 text-primary',
    );
  });
  it('should render BackgroundContainer with an image', () => {
    const container = render(<BackgroundContainer image />);
    expect(container.container.firstChild.firstChild).toHaveAttribute(
      'src',
      './images/main-bg.svg',
    );
    expect(container.container.firstChild.lastChild).toHaveClass('text-white');
  });
  it('should render BackgroundContainer with a background', () => {
    const container = render(<BackgroundContainer darkBg />);
    expect(container.container.firstChild.firstChild).toHaveClass('text-white');
    expect(container.container.firstChild).toHaveClass('bg-[#0A1128]');
  });
  it('should match a snapshot', () => {
    const container = render(<BackgroundContainer darkBg />);
    expect(container.container).toMatchSnapshot();
  });
});
