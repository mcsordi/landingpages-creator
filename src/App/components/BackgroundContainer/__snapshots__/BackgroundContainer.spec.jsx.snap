// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BackgroundContainer/> should match a snapshot 1`] = `
<div>
  <section
    class="bg-[#0A1128] flex flex-col w-full min-h-screen"
  >
    <div
      class="text-white relative mx-auto my-auto max-w-[1200px] px-8 py-6"
    />
  </section>
</div>
`;
